eng/sendtohelix.proj

```diff
@@ -81,23 +85,38 @@
     <_RuntimeInputs Include="$(TestHostRootPath)**/*.dll" />
   </ItemGroup>
 
+  <!-- Define which workloads require global tools. -->
+  <PropertyGroup>
+    <UseGlobalTools Condition="'$(UseGlobalTools)' == '' AND '$(Coverage)' == 'true'">true</UseGlobalTools>
+  </PropertyGroup>
+
   <!-- Add global tools to runtime -->
-  <ItemGroup Condition="'$(TargetsNetCoreApp)' == 'true'">
-    <TestArchiveRuntimeDependencies Include="$(GlobalToolsDir)**\*.*" />
+  <ItemGroup Condition="'$(UseGlobalTools)' == 'true'">
+    <TestArchiveRuntimeDependency Include="$(DotNetRoot)shared\Microsoft.NETCore.App\**\*.*" DestinationDir="shared\Microsoft.NETCore.App" />
+    <TestArchiveRuntimeDependency Include="$(DotNetRoot)host\**\*.*" DestinationDir="host" />
+    <TestArchiveRuntimeDependency Include="$(GlobalToolsDir)**\*.*" DestinationDir="tools" />
   </ItemGroup>
 
   <Target Name="CompressRuntimeDirectory"
-          Inputs="@(_RuntimeInputs);@(TestArchiveRuntimeDependencies)"
+          Inputs="@(_RuntimeInputs);@(TestArchiveRuntimeDependency)"
           Outputs="$(TestArchiveRuntimeFile)"
           Condition="'$(TargetGroup)' != 'AllConfigurations'">
 
     <!-- Copy additional test dependencies. -->
-    <Copy SourceFiles="@(TestArchiveRuntimeDependencies)"
-          DestinationFolder="$(TestHostRootPath)\%(RecursiveDir)"
+    <ItemGroup>
+      <_TestArchiveRuntimeDependency Include="@(TestArchiveRuntimeDependency)">
+        <DestinationDir>%(TestArchiveRuntimeDependency.DestinationDir)\%(RecursiveDir)</DestinationDir>
+      </_TestArchiveRuntimeDependency>
+    </ItemGroup>
+
+    <Copy SourceFiles="@(_TestArchiveRuntimeDependency)"
+          DestinationFiles="@(_TestArchiveRuntimeDependency -> '$([MSBuild]::NormalizePath('$(TestHostRootPath)', '%(DestinationDir)', '%(Filename)%(Extension)'))')"
           SkipUnchangedFiles="true"
           OverwriteReadOnlyFiles="true"
           Retries="3"
-          RetryDelayMilliseconds="300" />
+          RetryDelayMilliseconds="300">
+      <Output TaskParameter="DestinationFiles" ItemName="FileWrites" />
```

Comment:
Why do we need the Output?

Reply:
We don't need it as we usually don't call clean on sendtohelix.proj but it's the right thing to do in an msbuild sense.